lib/ansible/module_utils/basic.py

```diff
@@ -1260,9 +1260,16 @@ def log(self, msg, log_args=None):
             module = 'ansible-%s' % os.path.basename(__file__)
 
             # 6655 - allow for accented characters
-            if isinstance(msg, unicode):
-                # We should never get here as msg should be type str, not unicode
-                msg = msg.encode('utf-8')
+            # On Python 2 we need to pass byte strings to syslog/journal.
+            # On Python 3 we need to pass Unicode strings to syslog/journal.
+            if not isinstance(msg, str):
+                if isinstance(msg, unicode):
+                    # We should never get here as msg should be type str, not unicode
```

Comment:
Now that this method has been cut down I don't think this comment is necessarily true anymore.  best to omit it.